### PR TITLE
Add ForbidLooseMock phpstan rule

### PR DESCRIPTION
**Pull request by marmichalski**

### CODE REVIEW

File: src/Phpstan/Rule/ForbidLooseMock.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            return [];
+        }
+
+        if ($node->name->toString() !== 'createMock') {
+            return [];
+        }
+
+        $reflection = $scope->getClassReflection();
+        $parents = $reflection->getParentClassesNames();
+        if (!\in_array(TestCase::class, $parents, true)) {
```
Comment (marmichalski) [inline src/Phpstan/Rule/ForbidLooseMock.php:45]: bail if not in TestCase

File: src/Phpstan/Rule/ForbidLooseMock.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            return [];
+        }
+
+        if (
+            !\in_array(StrictMockingTestCase::class, $parents, true)
+            && !isset($reflection->getTraits(true)[StrictMockingTrait::class])
+        ) {
+            return [];
```
Comment (marmichalski) [inline src/Phpstan/Rule/ForbidLooseMock.php:53]: bail if not in StrictMockitty context, as those will be reported by the other rule

File: tests/unit/TestCaseTest.php [hunk 3]
```diff
@@ -17,6 +17,7 @@ final class TestCaseTest extends PHPUnitTestCase
     #[Test]
     public function mocks_created_require_methods_to_be_configured_explicitly(): void
     {
+        $this->createMock(\ArrayAccess::class);
```
Comment (marmichalski) [inline tests/unit/TestCaseTest.php:20]: 😠

File: src/Phpstan/Rule/ForbidLooseMock.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @implements Rule<MethodCall>
+ */
+final readonly class ForbidLooseMock implements Rule
+{
+    public function __construct() {}
```
Comment (ben-challis) [inline src/Phpstan/Rule/ForbidLooseMock.php:22]: redundant?

**Pull request merged**